### DISCUSSION TITLE
IK in Ragdoll

### DISCUSSION THREAD

**Discussion by motherd** [jrouwe/JoltPhysics.js#190] [Q&A] [answered]
I am currently using the ragdoll feature of Jolt Physics and would like to implement an IK system, such as "making the arm point towards a target." Does anyone know of a good approach for this?

I tried an implementation where I continuously applied forces using `AddImpulse` on a `Jolt.Body` to move the fingertip bone towards the IK target. While the fingertip bone does move to the target, the other bones linked to it break the positional constraints of `SixDOFConstraintSettings` and fly off in all directions.

If anyone has a solution or advice, I would appreciate your help.  
![image](https://github.com/user-attachments/assets/a9e4efc9-5734-4218-9cfa-30472aaef8ea)

**Comment by jrouwe** [comment, marked as answer]
Usually you pick one of these:
1. Use constraints with motors, drive the motors to the desired pose using e.g. `Ragdoll::DriveToPoseUsingMotors` or directly on the constraint
2. Use `Body::MoveKinematic` or `Ragdoll::DriveToPoseUsingKinematics` to move a body to an exact location (note that the body doesn't actually need to be kinematic, you're setting velocities to reach the desired pose)

**Reply by motherd** [reply]
Thank you for your advice.
I was able to achieve the desired behavior using `Body::MoveKinematic`, so I plan to proceed with this approach.
